README.md

```diff
@@ -48,20 +44,36 @@ by running it with the `--help` flag or reading the [rustc dev guide][rustcguide
 [gettingstarted]: https://rustc-dev-guide.rust-lang.org/getting-started.html
 [rustcguidebuild]: https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html
 
-### Building on a Unix-like system
-1. Make sure you have installed the dependencies:
-
-   * `g++` 5.1 or later or `clang++` 3.5 or later
-   * `python` 3 or 2.7
-   * GNU `make` 3.81 or later
-   * `cmake` 3.13.4 or later
-   * `ninja`
-   * `curl`
+### Dependencies
+
+Make sure you have installed the dependencies:
+
+   * `python` 3 or 2.7 (under the name python; python2 or python3 will not work)
    * `git`
-   * `ssl` which comes in `libssl-dev` or `openssl-devel`
+   * OpenSSL (`libssl-dev` or `openssl-devel` on most Unix distros)
+   * XZ (`liblzma-dev` on most distros)
```

Comment:
Do we really require libssl-dev and liblzma-dev?

For liblzma-dev, I would expect that the build script would just build it if it's not installed (https://docs.rs/crate/lzma-sys/0.1.20/source/build.rs). So it shouldn't be necessary.

For libssl-dev, I guess it's required for Cargo + bump-stage0, but most people won't be building Cargo, and it can be a bit of a painful dependency. So I think my preference for this list is to drop it or at least make clear that it's not strictly required.

```
$ cargo tree -i openssl-sys
openssl-sys v0.9.72
├── curl v0.4.44
│   ├── bump-stage0 v0.1.0 (/home/mark/Edit/rust/src/tools/bump-stage0)
│   ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   ├── crates-io v0.35.0 (/home/mark/Edit/rust/src/tools/cargo/crates/crates-io)
│   │   ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   │   └── cargo-test-support v0.1.0 (/home/mark/Edit/rust/src/tools/cargo/crates/cargo-test-support)
│   │       [dev-dependencies]
│   │       └── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   └── git2-curl v0.16.0
│       └── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
├── curl-sys v0.4.59+curl-7.86.0
│   ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   └── curl v0.4.44 (*)
├── git2 v0.15.0
│   ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   ├── cargo-test-support v0.1.0 (/home/mark/Edit/rust/src/tools/cargo/crates/cargo-test-support) (*)
│   └── git2-curl v0.16.0 (*)
├── libgit2-sys v0.14.0+1.5.0
│   ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
│   └── git2 v0.15.0 (*)
├── libssh2-sys v0.2.23
│   └── libgit2-sys v0.14.0+1.5.0 (*)
└── openssl v0.10.38
    └── crypto-hash v0.3.4
        └── cargo-util v0.2.3 (/home/mark/Edit/rust/src/tools/cargo/crates/cargo-util)
            ├── cargo v0.68.0 (/home/mark/Edit/rust/src/tools/cargo)
            └── cargo-test-support v0.1.0 (/home/mark/Edit/rust/src/tools/cargo/crates/cargo-test-support) (*)
```

Reply:
Oh huh, I guess we're only using it for rust-installer. I thought we linked it in rustc but it turns out I was thinking of zlib, not xz. https://github.com/rust-lang/rust/issues/74420

I'll remove liblzma and move libssl to its own section.